### ISSUE TITLE
`fail` is misleadingly called a boolean

### ISSUE THREAD

**Issue by ChrisJefferson** [gap-system/gap#827]
### Observed behaviour
```
gap> l := [];
gap> AppendList(l, fail);
Error, AppendList: <list2> must be a small list (not a boolean)
```
### Expected behaviour

Perhaps we should describe `fail` as `not "fail"`, rather than as a boolean (it is technically a boolean, but that's not necessarily obvious to users).

**Comment by fingolfin** [comment]
Or perhaps drop that "(not a FOO)" bit instead? I found that misleading in other instances in the past.

But perhaps it's been helpful for others?

**Comment by stevelinton** [comment]
The TNUM of fail is T_BOOL, so in that very low level sense and mechanistic sense it is a Boolean. Fixing this by special case in a lot of error handling code seems messy. We could change the name of that TNUM to "boolean or fail" which might be better.

**Comment by alex-konovalov** [comment]
I think we may merge #866. No tests in current packages has "not a boolean" example in their test files.

**Comment by fingolfin** [comment]
Fixed by #866
